### DISCUSSION TITLE
Update field value by adding value to it by record model in Framework function

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2326] [Q&A]
Through SDK its achievable by using + in field name for eg:
```
await pb.collection("products").update(id, {
    "stock+": 100,
})
```
But how to do it in custom function?
I tried below but its not working but i can do as ```product.Set("stock", product.GetFloat("stock")+100)``` and it works... but this may leads to error when multiple transaction take places.

```
product, err := txDao.FindRecordById("products", id)
if err != nil {
	return err
}
product.Set("stock+", 100)
if err := txDao.SaveRecord(product); err != nil {
	return err
}
```
How to achieve above?

**Comment by ganigeorgiev** [comment]
The field modifiers works only when using `RecordUpsert.LoadData`.

**But please note that the field modifier doesn't execute a transaction!** The `+` modifier is the same as `product.GetFloat("stock") + 100`.

> but this may leads to error when multiple transaction take places.

I'm not sure that I understand your concern. If the read and write operations are within a transaction, then it is guaranteed that no other write (eg. update of the "stock" field value) can take place until the transaction complete.

**Reply by iamdineshkumar** [reply]
Situation is when a purchase entry is made which will be stored in purchase collection at that time product stock need to be updated to the quantity. I am using OnRecordAfterCreateRequest to add stock of the product.
Here i need to do is
product.Set("stock+", purchasedQuanity)
Similar way i need to reduce stock while sale is saved using OnRecordAfterCreateRequest hook on sale collection
Here i need to do is
product.Set("stock+", saleQuanity)

So that stock will adjusted according to purchase and sale. But when i use above in even hooks its not giving those result.

**Reply by ganigeorgiev** [reply]
Again, if you already have the `FindRecordById` and the `SaveRecord` operations within a transaction, then it is guaranteed that no other write can take place and modify the product fields until the transaction is completed.

Or in other words, you don't have to worry for concurrent updates if the `product` item is fetched and modified within a transaction and you can safely use `product.Set("stock", product.GetFloat("stock")+100)`.
